stl/inc/chrono

```diff
@@ -2086,6 +2091,720 @@ namespace chrono {
+
+    struct _File_time_clock { // Implementation of trivial-clock
+        using rep                       = long long;
+        using period                    = chrono::system_clock::period;
+        using duration                  = chrono::duration<rep, period>;
+        using time_point                = chrono::time_point<_File_time_clock>;
+        static constexpr bool is_steady = false;
+
+        _NODISCARD static time_point now() noexcept { // get current time; undo epoch adjustment
+            return time_point(duration(_Xtime_get_ticks() + __std_fs_file_time_epoch_adjustment)); // TRANSITION, ABI
+        }
+
+#if _HAS_CXX20
+        // Assumes that FILETIME counts leap seconds only after the first 27 (i.e., after 1 January 2017), even though
+        // systems can opt out of this behavior.
+        static constexpr chrono::seconds _Skipped_filetime_leap_seconds{27};
+        static constexpr chrono::sys_days _Cutoff{
+            chrono::year_month_day{chrono::year{2017}, chrono::January, chrono::day{1}}};
+
+        template <class _Duration>
+        _NODISCARD static chrono::utc_time<common_type_t<_Duration, chrono::seconds>> to_utc(
+            const chrono::file_time<_Duration>& _File_time) {
+            using namespace chrono;
+            using _CT = common_type_t<_Duration, seconds>;
+            const auto _Ticks =
+                _File_time.time_since_epoch() - duration_cast<seconds>(duration{__std_fs_file_time_epoch_adjustment});
+
+            if (_Ticks < _Cutoff.time_since_epoch()) {
+                return utc_clock::from_sys(sys_time<_CT>{_Ticks});
+            } else {
+                return utc_time<_CT>{_Ticks + _Skipped_filetime_leap_seconds};
+            }
+        }
+
+        template <class _Duration>
+        _NODISCARD static chrono::file_time<common_type_t<_Duration, chrono::seconds>> from_utc(
+            const chrono::utc_time<_Duration>& _Utc_time) {
+            using namespace chrono;
+            using _CT = common_type_t<_Duration, seconds>;
+            file_time<_CT> _File_time{duration_cast<seconds>(duration{__std_fs_file_time_epoch_adjustment})};
+
+            if (_Utc_time < utc_seconds{_Cutoff.time_since_epoch()} + _Skipped_filetime_leap_seconds) {
+                _File_time += utc_clock::to_sys(_Utc_time).time_since_epoch();
+            } else {
+                _File_time += _Utc_time.time_since_epoch() - _Skipped_filetime_leap_seconds;
+            }
+
+            return _File_time;
+        }
+#endif // ^^^ _HAS_CXX20
+    };
+} // namespace filesystem
+#endif // ^^^ _HAS_CXX17
+
+namespace chrono {
+#if _HAS_CXX20
+    // [time.clock.conv]
+
+    template <class _DestClock, class _SourceClock>
+    struct clock_time_conversion;
+
+    // [time.clock.cast.id]
+
+    template <class _Clock>
+    struct clock_time_conversion<_Clock, _Clock> {
+        template <class _Duration>
+        _NODISCARD auto operator()(const time_point<_Clock, _Duration>& _Time) const

```

Comment:
Done, though I'm not sure if you'd also prefer to see the trailing return type for the ones that are depicted as `auto`, e.g.
```cpp
template<class SourceClock>
struct clock_time_conversion<system_clock, SourceClock> {
  template<class Duration>
    auto operator()(const time_point<SourceClock, Duration>& t) const
      -> decltype(SourceClock::to_sys(t));
}
```

Reply:
I think in general, we like to mirror what's in the Standard as much as possible unless there is significant performance/code duplication avoidance/etc. gains to be made (this eases review and reduces the probability of bugs).  

Since these are depicted with `auto` and trailing return type in the Standard, I'm fine with `auto` in their definition.